src/rgw/rgw_data_sync.cc

```diff
@@ -981,13 +997,15 @@ class RGWListBucketIndexesCR : public RGWCoroutine {
           int shard_id = (int)iter->first;
           rgw_data_sync_marker& marker = iter->second;
           marker.total_entries = entries_index->get_total_entries(shard_id);
+          objv.clear();
           spawn(new RGWSimpleRadosWriteCR<rgw_data_sync_marker>(
 		  dpp, sync_env->async_rados, sync_env->svc->sysobj,
 		  rgw_raw_obj(sync_env->svc->zone->get_zone_params().log_pool,
 			      RGWDataSyncStatusManager::shard_obj_name(
 				sc->source_zone, shard_id)),
-		  marker),
+		  marker, &objv),
 		true);
+          objvs[shard_id] = objv;
```

Comment:
we can pass `&objvs[shard_id]` directly to `RGWSimpleRadosWriteCR` here. and we shouldn't call `clear()` before this write - we need to use the same objv from our earlier call to `RGWReadDataSyncStatusCoroutine` or `RGWInitDataSyncStatusCoroutine`

Reply:
fixed